conbench/api/_google.py

```diff
@@ -1,44 +1,72 @@
 import json
+import logging
 import os
+import time
 
 import flask as f
 import requests
 
+from ..config import Config
 
-def get_google_config():
+log = logging.getLogger(__name__)
+
+# Rely on INTENDED_BASE_URL to have a trailing slash.
+OIDC_CALLBACK_URL = Config.INTENDED_BASE_URL + "api/google/callback"
+
+
+def get_oidc_config():
     client_id = os.environ.get("GOOGLE_CLIENT_ID", None)
     client_secret = os.environ.get("GOOGLE_CLIENT_SECRET", None)
-    discovery_url = "https://accounts.google.com/.well-known/openid-configuration"
+    discovery_url = Config.OIDC_ISSUER_URL + "/.well-known/openid-configuration"
     return discovery_url, client_id, client_secret
 
 
-def get_google_client():
+def get_oidc_client():
     from oauthlib.oauth2 import WebApplicationClient
 
-    discovery_url, client_id, _ = get_google_config()
-    google = requests.get(discovery_url).json()
+    discovery_url, client_id, _ = get_oidc_config()
+
+    # Pragmatic healing for transient errors. Better: cache OP config across
+    # login requests. Also, if we want to retry here in the future: consider
+    # using tenacity.
+    for attempt in range(4):
+        try:
+            oidc_provider_config = requests.get(discovery_url).json()
+            break
+        except requests.exceptions.RequestException as exc:
+            log.info("err getting OP config (attempt %s): %s -- retry", attempt, exc)
+            time.sleep(2)
+
     client = WebApplicationClient(client_id)
-    return client, google
+    return client, oidc_provider_config
 
 
 def auth_google_user():
-    client, google = get_google_client()
-    redirect_uri = f.url_for("api.callback", _external=True, _scheme="https")
```

Comment:
For clarity, instead of

```python
redirect_uri = f.url_for("api.callback", _external=True, _scheme="https")
```
we now do
```python
Config.INTENDED_BASE_URL + "api/google/callback"
```
at the top of this module.

Reply:
The `api/google` part of that URL maybe should at some point change into something less googly for cosmetic reasons.